services/web/server/tests/unit/with_dbs/01/test_comp_tasks_listening_task.py

```diff
@@ -125,17 +125,17 @@ async def test_listen_comp_tasks_task(
         # tests whether listener gets hooked calls executed
         for call_name, mocked_call in mock_project_subsystem.items():
             if call_name in expected_calls:
-                for attempt in _retry_if_fails():
+                async for attempt in _async_retry_if_fails():
                     with attempt:
                         mocked_call.assert_awaited()
 
             else:
                 mocked_call.assert_not_called()
 
 
-def _retry_if_fails():
-    # Helper that retries to acount for some uncontrolled delays
-    return tenacity.Retrying(
+def _async_retry_if_fails():
```

Comment:
minor: to be honest I still prefer the decorator version. and it automatically handles the async/sync paradigm.

Reply:
I guess is a matter of taste ... at the end of the day, is +/- the same number of lines

I prefer
```python
for attempt in Retrying(..):
   with attempt:
       do_it()
```
and you prefer
```python
@retry(...)
def do_it_tenacitous():
   do_it()
_do_it_tenacitous()
```